### PR TITLE
Module10

### PR DESCRIPTION
**Pull request by avdhut1977**

### CODE REVIEW

File: todo_app/app.py [hunk 1]
```diff
@@ -1,53 +1,126 @@
...
+            return ROLE
+        else:            
+            return 'readonly'
+
+    def read_write_user():
+        user_role = get_user_role()
+        if (user_role == ROLE):
+            return True 
+
+        return False
```
Comment (BenRamchandani) [inline todo_app/app.py]: Same as
```suggestion
        return user_role == Role
```

File: todo_app/app.py [hunk 2]
```diff
@@ -1,53 +1,126 @@
...
+    @login_required
     def add_todo_task():
-        taskDb.create_todo_task(request.form.get('title'))
+        if (read_write_user()):
+            taskDb.create_todo_task(request.form.get('title'))
+            return redirect('/')
+        else:
+            return abort(403)
+        
         return redirect('/')
```
Comment (BenRamchandani) [inline todo_app/app.py:87]: This return is dead code now

File: todo_app/view_model.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import datetime
 
 class ViewModel:     
-    def __init__(self, tasks):   
+    def __init__(self, tasks,role='readonly'):   
         self._tasks = tasks
```
Comment (BenRamchandani) [inline todo_app/view_model.py:5]: You need to store `role` on `self` - at the moment it's just ignored!

File: .travis.yml [hunk 4]
```diff
@@ -21,5 +21,5 @@ deploy:
   script:    
     - heroku container:release web --app $HEROKU_APP
   on:
-    branch: master
+    branch: module10
```
Comment (BenRamchandani) [inline .travis.yml]: Remember to change this back when you merge 🙂

**Review by BenRamchandani — COMMENTED** [review]
You're missing a `self.role = role` in `view_model.py`, but once that's fixed it all works as expected.

Make sure you fix that before merging, but once that's done it all looks good.

File: todo_app/app.py (hunk 1 above)
Comment (avdhut1977) [reply todo_app/app.py]: Done

File: todo_app/view_model.py (hunk 3 above)
Comment (avdhut1977) [reply todo_app/view_model.py:5]: Done Thanks

File: .travis.yml (hunk 4 above)
Comment (avdhut1977) [reply .travis.yml]: Done

**Pull request merged**